test/generators/suspenders/factory_bot_generator_test.rb

```diff
@@ -0,0 +1,215 @@
+        Bundler.stubs(:with_unbundled_env).yields
+        generator.expects(:run).with("bundle install").once
+
+        capture(:stdout) do
+          generator.add_factory_bot
+        end
+      end
+
+      test "removes fixture definitions" do
+        File.open(app_root("test/test_helper.rb"), "w") { _1.write test_helper }
+
+        run_generator
+
+        assert_file app_root("test/test_helper.rb") do |file|
+          assert_match(/# fixtures :all/, file)
+        end
+      end
+
+      test "adds gem to Gemfile" do
+        run_generator
+
+        assert_file app_root("Gemfile") do |file|
+          assert_match(/group :development, :test do\n  gem "factory_bot_rails"\nend/, file)
+        end
+      end
+
+      test "includes syntax methods" do
+        File.open(app_root("test/test_helper.rb"), "w") { _1.write test_helper }
+
+        run_generator
+
+        assert_file app_root("test/test_helper.rb") do |file|
+          assert_match(/class TestCase\n    include FactoryBot::Syntax::Methods/, file)
+        end
+      end
+
+      test "creates definition file" do
+        run_generator
+
+        assert_file app_root("test/factories.rb") do |file|
+          assert_match definition_file, file
+        end
+      end
+
+      test "creates linting test" do
+        run_generator
+
+        assert_file app_root("test/factory_bots/factories_test.rb") do |file|
+          assert_match factories_test, file
+        end
+      end
+
+      test "creates dev prime task" do
+        run_generator
+
+        assert_file app_root("lib/tasks/dev.rake") do |file|
+          assert_match dev_rake, file
+        end
+      end
+
+      private
+
+      def prepare_destination
+        mkdir "test"
+        touch "Gemfile"
+      end
+
+      def restore_destination
+        remove_dir_if_exists "test"
+        remove_dir_if_exists "spec"
+        remove_file_if_exists "Gemfile"
+        remove_dir_if_exists "lib/tasks"
+      end
+
+      def test_helper
+        <<-RUBY
+    ENV["RAILS_ENV"] ||= "test"
+    require_relative "../config/environment"
+    require "rails/test_help"
+
+    module ActiveSupport
+      class TestCase
+        # Run tests in parallel with specified workers
+        parallelize(workers: :number_of_processors)
+
+        # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
+        fixtures :all
+
+        # Add more helper methods to be used by all tests here...
+      end
+    end
+        RUBY
+      end
+
+      def factories_test
+        <<~RUBY
+          require "test_helper"
+
+          class FactoryBotsTest < ActiveSupport::TestCase
+            class FactoryLintingTest < FactoryBotsTest
+              test "linting of factories" do
+                FactoryBot.lint traits: true
+              end
+            end
+          end
+        RUBY
+      end
+
+      def definition_file
+        <<~RUBY
+          FactoryBot.define do
+          end
+        RUBY
+      end
```

Comment:
I guess things like this could be placed in the `fixtures` directory, but for now I figured I'd just keep them inline. Maybe they should actually be set in the setup phase of the associated tests thought?

Reply:
I ended up moving these inline: 2bdf0d3cb21e13ea4ff4c558b55289e696910b69